src/base/bittorrent/private/sequentialdownloadsguard.h

```diff
@@ -0,0 +1,58 @@
+#ifndef SEQUENTIALDOWNLOADGUARD_H
+#define SEQUENTIALDOWNLOADGUARD_H
+
+#include <QObject>
+
+class QTimer;
+
+namespace BitTorrent
+{
+    class Session;
+    class TorrentHandle;
+
+    /**
+       @brief Enable / Disable Sequential download by torrents availability.
+
+       Enable / Disable Sequential download by torrents availability, torrent has to be previewable
+       to enable sequential download.
+       Sequential download can be enabled globally, in main options and per torrent
+       in Add new Torrent dialog.
+     */
+    class SequentialDownloadsGuard : public QObject
+    {
+        Q_OBJECT
+
+    public:
+        explicit SequentialDownloadsGuard(Session *const session, QTimer *const sequentialGuardTimer);
+
+        static const int SEQ_DOWNLOAD_FIRST_REFRESH_INTERVAL = 5 * 1000; // 5sec
+
+    public slots:
+        void recheckSequentialDownloads();
+
+    private:
+        void updateSequentialDownload(TorrentHandle *const torrentHandle);
+        bool isSequentialDownloadEnabled(TorrentHandle *const torrentHandle) const;
+        void increaseTimerInterval();
+        void enableSequentialDownload(TorrentHandle *const torrentHandle);
+        void disableSequentialDownload(TorrentHandle *const torrentHandle);
+        void logResult(TorrentHandle *const torrentHandle) const;
+
+        Session *m_session;
+        QTimer *m_sequentialGuardTimer;
+        bool m_intervalIncreased = false;
+
+        /**
+           @brief If availability is higher than this value, sequential download for a torrent
+           will be enabled.
+         */
+        static const int AVAILABILITY_LIMIT = 10;
```

Comment:
@evsh:  I was specifically referring to adding an option for the user to change the value set by this line of code (which determines the minimum `availability` threshold before a torrent can use sequential downloading).  

To clarify my prior debate with @silverqx, I was saying that so long as I can change this value myself (be it via the GUI or an environment-variable), I have no problem with the changes introduced by this PR.  On the other hand, I DO NOT want to be locked to 10 exclusively, because I currently rely on sequential downloading to balance the network strain caused by qBittorrent (which sequential downloading does surprisingly well, as I discovered years ago).